packages/synapse-react-client/src/components/AccessRequirementList/ManagedACTAccessRequirementRequestFlow/DataAccessRequestAccessorsFilesForm.tsx

```diff
@@ -145,17 +154,81 @@ export default function DataAccessRequestAccessorsFilesForm(
     })
   }
 
-  const { mutate: submit } = useSubmitDataAccessRequest({
-    onSuccess: () => {
-      onSubmissionCreated()
-    },
-    onError: onError,
-  })
+  const { mutate: submit, isLoading: isLoadingSubmitDataAccessRequest } =
+    useSubmitDataAccessRequest({
+      onSuccess: () => {
+        onSubmissionCreated()
+      },
+      onError: onError,
+    })
 
-  const { mutateAsync: updateRequestAsync } = useUpdateDataAccessRequest({
+  const {
+    mutate: updateRequest,
+    mutateAsync: updateRequestAsync,
+    isLoading: isLoadingUpdateDataAccessRequest,
+  } = useUpdateDataAccessRequest({
     onError: onError,
   })
 
+  const isLoading =
+    isLoadingGetDataAccessRequest ||
+    isLoadingUpdateDataAccessRequest ||
+    isLoadingSubmitDataAccessRequest
+
+  /**
+   * This effect comprises a collection of updates we should immediately apply to a data access request.
+   */
+  useEffect(() => {
```

Comment:
These transformations were previously applied in the `select` option in `useGetDataAccessRequestForUpdate`. In that scenario, we didn't push the updates to the server. Now, we track if an update should be made, and if so, push that update to the server.